src/date/expand-pattern/augment-format.js

```diff
@@ -4,30 +4,42 @@ define([
 	"../../util/string/repeat"
 ], function( dateExpandPatternNormalizePatternType, datePatternRe, stringRepeat ) {
 
-return function( requestedSkeleton, bestMatchFormat ) {
-	var i, j, matchedType, matchedLength, requestedType, requestedLength,
+// See: http://www.unicode.org/reports/tr35/tr35-dates.html#Matching_Skeletons
+return function( requestedSkeleton, bestMatchFormat, decimalSeparator ) {
+	var i, j, bestMatchFormatParts, countOfFractionalSeconds, fractionalSecondMatch, matchedType,
+		matchedLength, requestedType, requestedLength, requestedSkeletonParts,
+		skeletonWithoutFractionalSeconds,
 
 		// Using an easier to read variable.
 		normalizePatternType = dateExpandPatternNormalizePatternType;
 
-	requestedSkeleton = requestedSkeleton.match( datePatternRe );
-	bestMatchFormat = bestMatchFormat.match( datePatternRe );
+	fractionalSecondMatch = requestedSkeleton.match( /S/g );
+	countOfFractionalSeconds = fractionalSecondMatch ? fractionalSecondMatch.length : 0;
+	skeletonWithoutFractionalSeconds = requestedSkeleton.replace( /S/g, "" );
 
-	for ( i = 0; i < bestMatchFormat.length; i++ ) {
-		matchedType = bestMatchFormat[i].charAt( 0 );
-		matchedLength = bestMatchFormat[i].length;
-		for ( j = 0; j < requestedSkeleton.length; j++ ) {
-			requestedType = requestedSkeleton[j].charAt( 0 );
-			requestedLength = requestedSkeleton[j].length;
+	requestedSkeletonParts = skeletonWithoutFractionalSeconds.match( datePatternRe );
```

Comment:
Yeah, that makes sense. Is it cool if I keep both of those functions in `augment-format.js`? That way the file perfectly reflects the "adjustment" section of the [spec](http://www.unicode.org/reports/tr35/tr35-dates.html#Missing_Skeleton_Fields): 

> Once a skeleton match is found, the corresponding pattern is used, but with adjustments. Consider the following dateFormatItem:
> 
>     <dateFormatItem id="yMMMd">d MMM y</dateFormatItem>
> If this is the best match for yMMMMd, pattern is automatically expanded to produce the pattern "d MMMM y" in response to the request. Of course, if the desired behavior is that a request for yMMMMd should produce something other than "d MMMM y", a separate dateFormatItem must be present, for example:
> 
>     <dateFormatItem id="yMMMMd">d 'de' MMMM 'de' y</dateFormatItem>
> If the requested skeleton included both seconds and fractional seconds and the dateFormatItem skeleton included seconds but not fractional seconds, then the seconds field of the corresponding pattern should be adjusted by appending the locale’s decimal separator, followed by the sequence of ‘S’ characters from the requested skeleton.